node/src/service/p2p/bootstrap.rs

```diff
@@ -61,46 +105,197 @@ impl BootstrapConfig {
 pub(crate) async fn bootstrap(
     mut swarm: Swarm<BootstrapBehaviour>,
     addr: Multiaddr,
+    bootstrap_addresses: Vec<Multiaddr>,
 ) -> Result<(), P2PError> {
     info!("Starting P2P bootstrap node at {addr}");
     swarm.listen_on(addr)?;
+    for addr in bootstrap_addresses {
+        info!("Attempting to dial peer at {addr}");
+        if swarm.dial(addr.clone()).is_err() {
+            warn!("Failed to dial peer");
+        }
```

Comment:
```suggestion
    for addr in bootstrap_addresses {
        info!("Attempting to dial peer at {addr}");
        if swarm.dial(addr.clone()).is_err() {
            warn!("Failed to dial peer: {}", addr);
        }
```

Btw, what happens if a bootstrap node later joins a network? Will it still be like connected/connect somehow to the bootstrap nodes and still exchange information?

Reply:
The connection is established when one of the nodes connect to each other. So if a node is already running and another one dials (connects) the running node the information will still be shared.
Note: This PR does not implement any catch-up or recovery, meaning if a bootstrap node joins late, after registrations have happened, it will have incomplete data.
To keep this PR sizeable I have not implemented this in here but this is certainly an improvement we have to make.